Update references to SDKv2 to use plugin-framework (#563)
hashicorp/terraform-docs-common: website/docs/docs/partnerships.mdx

### 1. Prepare

To get started with the Terraform provider development, we recommend reviewing and following the resources listed below.

#### Provider Development Kit

a) Writing custom providers [tutorial](/terraform/tutorials/providers-plugin-framework)

b)  Get started with the Plugin Framework [repository] and the writing custom providers [guide](/terraform/plugin/framework).

> c) Creating a Terraform Provider with the Plugin Framework: [video](https://www.youtube.com/watch?v=16qs7LJSyps)

Reviewer: ```suggestion
c) Creating a Terraform Provider with the Plugin Framework following our video guide:
<VideoEmbed url="https://www.youtube.com/watch?v=16qs7LJSyps"/>
```

> d) An overview of the new Plugin Framework and improvements to the Terraform Provider ecosystem: [video](https://www.youtube.com/watch?v=pAeQC0fC3Xw)

Reviewer: ```suggestion
d) An overview of the new Plugin Framework and improvements to the Terraform Provider ecosystem:
<VideoEmbed url="https://www.youtube.com/watch?v=pAeQC0fC3Xw"/>
```

### 1. Prepare

To get started with the Terraform provider development, we recommend reviewing and following the resources listed below.

#### Provider Development Kit

> a) Write your own custom provider with our [tutorial](/terraform/tutorials/providers-plugin-framework).

Reviewer: ```suggestion
a) Write your own custom provider with our [tutorials](/terraform/tutorials/providers-plugin-framework).
```

b) Get started with the Plugin Framework [repository](https://github.com/hashicorp/terraform-provider-scaffolding-framework), and reference the [documentation](/terraform/plugin/framework).

c) Creating a Terraform Provider with the Plugin Framework following our video guide:
<VideoEmbed url="https://www.youtube.com/watch?v=16qs7LJSyps"/>

d) An overview of the new Plugin Framework and improvements to the Terraform Provider ecosystem:
<VideoEmbed url="https://www.youtube.com/watch?v=pAeQC0fC3Xw"/>

e) Example provider for reference: [HTTP](https://github.com/hashicorp/terraform-provider-http)

> f) Contributing to Terraform [guidelines](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md)

Reviewer: ```suggestion
f) Contribute to Terraform's [guidelines](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md)
```

> g) HashiCorp Developer Discuss [forum](https://discuss.hashicorp.com/c/terraform-providers/tf-plugin-sdk/43)

Reviewer: ```suggestion
g) Discuss providers in the HashiCorp developer discuss [forum](https://discuss.hashicorp.com/c/terraform-providers/tf-plugin-sdk/43).
```

#### Provider Development Kit

a) Writing custom providers [guide](/terraform/plugin/framework)

b) Creating a Terraform Provider with the Plugin Framework: [video](https://www.youtube.com/watch?v=16qs7LJSyps)

c) An overview of the new Plugin Framework and improvements to the Terraform Provider ecosystem: [video](https://www.youtube.com/watch?v=pAeQC0fC3Xw)

> d) Example provider for reference: [HTTP](https://github.com/hashicorp/terraform-provider-http)

Author: yes, we should lol. thanks for the suggestion, will update!
